src/components/ProductDetails/Details.jsx

```diff
@@ -0,0 +1,40 @@
+import React, { useEffect } from "react";
+import { useSelector } from "react-redux";
+import { useParams } from "react-router";
+import useCartHooks from "../../redux/Shopping/shopping-actions";
+
+
+const Details = () => {
+  const current = useSelector(state => state.shop.currentItem);
+  const { addToCart, loadCurrentItem } = useCartHooks()
+  const params = useParams();
+
+  useEffect(() => {
+    if (!current) {
+      loadCurrentItem(Number(params.id))
+    }
+  }, [params]);
```

Comment:
dependency should be params.id as it only depends on id. Also I'd recommend detructure params instead of using dot operator

Reply:
Done